### PR TITLE
fix(storage): allow to use age=0 in OLM conditions

### PR DESCRIPTION
**Pull request by frankyn**
Storage Go doesn't allow setting `AgeInDays=0` after discussing with @tritone. This PR is the agreed upon path to support `Age=0` in OLM conditions. It also notes that other int fields in OLM Conditions do not support `0` values.

Fixes: #6539, #6240

### CODE REVIEW

File: storage/bucket.go
Comment (tritone) [inline storage/bucket.go:702]: ... setting AgeInDays to 0.

File: storage/bucket.go
Comment (tritone) [inline storage/bucket.go:1639]: Update this in `toLifecycleFromProto` below as well.

**Review by tritone — COMMENTED** [review]
Thanks Frank!

File: storage/bucket.go
Comment (tritone) [inline storage/bucket.go:724]: "will be ignored", not "will not be ignored". Same in L729 below.

Also, add this note to NumNewerVersions as well.

**Review by tritone — COMMENTED** [review]
Some comment verbiage stuff. Also, would you mind adding a case to integration tests as well? Could go here: https://github.com/googleapis/google-cloud-go/blob/main/storage/integration_test.go#L255

**Comment by frankyn** [comment]
Thanks @tritone, I'm blocked at the moment. I'm not sure how I can detect if Age is actually returned by the API as `0` or if it's not. 

Do you have tips or guidance here? I tried using NullFields but that's only for Requests and not Responses.

**Comment by tritone** [comment]
Update: conferred with @frankyn and we see that LifecycleRule.Age is implemented as `int64` in google-api-go-client as well. Therefore, we can't see from the response whether the server sent the field or not if it was a zero. This means that we could end up corrupting users' lifecycle rules if they use this library to do a get/modify/update loop on their set of rules for a bucket.

Really this field should have been special-cased in the Apiary library as a pointer, but we can't make that change now in Apiary as it would be a breaking change for Apiary users. We're looking into workarounds and should have an update soon.

File: storage/bucket.go
Comment (tritone) [inline storage/bucket.go:1559]: I would say `if *rr.Condition.Age == 0`, no need to create another pointer. (Actually I don't know if this works as written, since the pointers will not actually be equal.)

File: storage/bucket.go
Comment (tritone) [inline storage/bucket.go:1608]: We probably need to do some deeper digging into how this works/doesn't for gRPC.

**Review by tritone — COMMENTED** [review]
As I mentioned before, I'd like to have a case added to the integration tests for this as well, showing that the roundtrip of having allObjects set to/from the server works as expected. Can go here: https://github.com/googleapis/google-cloud-go/blob/main/storage/integration_test.go#L255

File: storage/bucket.go
Comment (frankyn) [reply storage/bucket.go:1608]: Is there a place where metadata integration tests exist, maybe I'm not looking hard enough? Also this could wait until after this PR is merged unless there's a reason to wait to sus out gRPC first?

**Review by tritone — APPROVED** [review]

File: storage/bucket.go
Comment (frankyn) [inline storage/bucket.go:1457]: Question: we may want to return an error here when both `AgeInDays` and `AllObjects` are set at the same time, what do y'all think?

**Review by codyoss — APPROVED** [review]
Overall LGTM, just a small nit.

File: storage/bucket.go
Comment (tritone) [reply storage/bucket.go:1456]: Discussed offline; AllObjects should take precedence if both are set.

**Review by tritone — APPROVED** [review]

File: storage/bucket.go
Comment (tritone) [reply storage/bucket.go:1609]: Yes, this can happen later

**Pull request merged**
